tests/integration/bond_test.py

```diff
@@ -353,6 +355,73 @@ def test_changing_slave_order_keeps_mac_of_existing_bond(bond99_with_2_slaves):
     )
 
 
+@pytest.mark.tier1
+def test_adding_a_slave_keeps_mac_of_existing_bond(bond99_with_eth2, eth1_up):
+    bond_state = bond99_with_eth2[Interface.KEY][0]
+    bond_slaves = bond_state[Bond.CONFIG_SUBTREE][Bond.SLAVES]
+    bond_slaves.insert(0, ETH1)
+
+    current_state = statelib.show_only((BOND99,))
+
+    libnmstate.apply(bond99_with_eth2)
+    modified_state = statelib.show_only((BOND99,))
+    assert_mac_address(
+        modified_state, current_state[Interface.KEY][0][Interface.MAC]
+    )
+
+
+@pytest.mark.tier1
+def test_adding_slaves_to_empty_bond_keeps_mac(eth1_up):
+    with bond_interface(BOND99, []) as state:
+        bond_state = state[Interface.KEY][0]
+        bond_state[Bond.CONFIG_SUBTREE][Bond.SLAVES] = [ETH1]
+
+        current_state = statelib.show_only((BOND99,))
+
+        libnmstate.apply(state)
+        modified_state = statelib.show_only((BOND99,))
+        with pytest.raises(AssertionError):
+            assert_mac_address(
+                modified_state, current_state[Interface.KEY][0][Interface.MAC]
+            )
+
+
+@pytest.mark.tier1
+def test_replacing_slaves_keeps_mac_of_existing_bond(
+    bond99_with_eth2, eth1_up
+):
+    bond_state = bond99_with_eth2[Interface.KEY][0]
+    bond_state[Bond.CONFIG_SUBTREE][Bond.SLAVES] = [ETH1]
+
+    current_state = statelib.show_only((BOND99,))
+
+    libnmstate.apply(bond99_with_eth2)
+    modified_state = statelib.show_only((BOND99,))
+    assert_mac_address(
+        modified_state, current_state[Interface.KEY][0][Interface.MAC]
+    )
+
+
+@pytest.mark.xfail(
+    raises=NmstateVerificationError,
+    reason="https://github.com/nmstate/nmstate/issues/932",
+)
+@pytest.mark.parametrize("slaves", ([ETH2], []))
+def test_removing_slaves_keeps_mac_of_existing_bond(
```

Comment:
Does this work with plain `ip link` or sysfs commands?

Reply:
The problem is , right now we are not able to remove all the slaves from a linux bond. It is a nmstate issue. I've not tried it with sysfs or `ip link`

Reply:
Tested with `ip link`. It is not keeping the mac address.